### ISSUE TITLE
Query Service - Cache-Control header for static assets

### ISSUE THREAD

**Issue by tiffon** [jaegertracing/jaeger#616]
Add the `Cache-Control: public, immutable, max-age=31536000` header to all non-html static assets served by the query service. Mainly, to all static assets **except `index.html`**.

**With the exception of index.html**, the static assets generated by the UI can safely be cached because they all have part of their hash in the filename.

Currently, things are pretty quick because the query-service tends to return 304, so this won't be a game-changer, but it will trim some time off.

**Comment by yurishkuro** [comment]
including images?

**Comment by tiffon** [comment]
Yes, including images (as long as they are referenced in JS, which they are).

**Comment by yurishkuro** [comment]
Here's one example that's not salted: `<link rel="shortcut icon" href="/favicon.ico">`

**Comment by tiffon** [comment]
@yurishkuro Great catch! I'll make a note of it for the path prefix measures, too...

Maybe we can apply the updated Cache-Control header to anything with a hash in the filename?

Testing the request URL/path with a regex along the lines of `[^/]\.[a-f0-9]{6,}\.\w{2,8}$` might work.

**Comment by jpkrohling** [comment]
Should this be closed?
